packages/react-codemirror/src/lang-cypher/autocomplete.ts

```diff
@@ -65,17 +83,70 @@ export const cypherAutocomplete: (config: CypherConfig) => CompletionSource =
       context.explicit,
     );
 
-    return {
-      from: context.matchBefore(/(\w|\$)*$/).from,
-      options: options.map((o) => ({
-        label: o.label,
-        type: completionKindToCodemirrorIcon(o.kind),
-        apply:
-          o.kind === CompletionItemKind.Snippet
-            ? // codemirror requires an empty snippet space to be able to tab out of the completion
-              snippet((o.insertText ?? o.label) + '${}')
-            : undefined,
-        detail: o.detail,
-      })),
-    };
+    if (config.featureFlags.signatureInfoOnAutoCompletions) {
+      return {
+        from: context.matchBefore(/(\w|\$)*$/).from,
+        options: options.map((o) => {
+          let maybeInfo = {};
+          const newDiv = document.createElement('div');
+
+          if (o.signature) {
+            const header = document.createElement('p');
+            header.setAttribute('class', 'cm-completionInfo-signature');
+            header.textContent = o.signature;
+            newDiv.appendChild(header);
+          }
+
+          if (o.documentation) {
+            const paragraph = document.createElement('p');
+            paragraph.textContent = getDocString(o.documentation);
+            newDiv.appendChild(paragraph);
+          }
+
+          maybeInfo = {
+            info: () =>
+              new Promise((resolve) => {
+                resolve(newDiv);
+              }),
```

Comment:
Assuming this needs to a promise, the easiest way/clearest way to wrap it is to just do 
```suggestion
              Promise.resolve(newDive)
```